pkg/serviceaccount/jwt.go

```diff
@@ -177,7 +177,7 @@ func (j *jwtTokenAuthenticator) AuthenticateToken(tokenData string) (user.Info,
 		return nil, false, err
 	}
 
-	return sa.UserInfo(), true, nil
+	return &authenticator.Response{User: sa.UserInfo()}, true, nil
```

Comment:
needs a TODO to verify the token audience and echo in the response if ctx has an audience limitation

Reply:
I think this (and the other related TODOs) means most of the authenticators will need to know the implicit apiserver audiences at construction time. If the token has no audience (legacy service account token), then we should accept AuthenticateToken requests for the implicit apiserver audiences.